src/Contracts/Str.php

```diff
@@ -0,0 +1,46 @@
+<?php
```

Comment:
`Str` should be a primitive type. I don't see much value in making it implement a interface, as there shouldn't be multiple `Str` implementations. Specially so when this is only a wrapper for standard functionality.

Maybe if we intend to implement both immutable (which in this context should be akin to constant) strings, it would make sense to have them both implement a contract. Maybe I don't care if my string is mutable. I'm not sure if it makes sense.